python/MoltresXS.py

```diff
@@ -0,0 +1,280 @@
+
+
+class serpent_xs:
+    """
+        Python class that reads in a serpent res file and organizes the cross section data
+        into a numpy dictionary. Currently set up to read an arbitrary number of energy groups,
+        an arbitrart number of delayed neutron groups, an arbitrary number of identities,
+        an arbitrary number of temperature branches, and an arbitrary number of burnups.
+
+        Parameters
+        ----------
+        FileIn: str
+            Name of file containing collapsed cross section data
+        Returns
+        ----------
+        xs_lib: dict
+            A hierarchical dictionary, organized by burnup, id, and temperature.
+            Currently stores REMXS, FISSXS, NSF, FISSE, DIFFCOEF, RECIPVEL, CHI, BETA_EFF, DECAY_CONSTANT and
+            GTRANSFXS.
+    """
+
+    def __init__(self, FileIn):
+        data = serpent.parse_res(FileIn)
+        try:
+            num_burn = len(np.unique(data['BURNUP'][:][0]))
+        except(KeyError):
+            num_burn = 1
+        num_uni = len(np.unique(data['GC_UNIVERSE_NAME']))
+        num_temps = int(len(data['GC_UNIVERSE_NAME'])/(num_uni*num_burn))
+        self.xs_lib = {}
+        for i in range(num_burn):
+            self.xs_lib[i] = {}
+            for j in range(num_uni):
+                self.xs_lib[i][j] = {}
+                for k in range(num_temps):
+                    self.xs_lib[i][j][k] = {}
+                    index = i*(num_uni)+k*(num_burn*num_uni)+j
+                    self.xs_lib[i][j][k]["REMXS"] = data['INF_REMXS'][index][::2]
+                    self.xs_lib[i][j][k]["FISSXS"] = data['INF_FISS'][index][::2]
+                    self.xs_lib[i][j][k]["NSF"] = data['INF_NSF'][index][::2]
+                    self.xs_lib[i][j][k]["FISSE"] = data['INF_KAPPA'][index][::2]
+                    self.xs_lib[i][j][k]["DIFFCOEF"] = data['INF_DIFFCOEF'][index][::2]
+                    self.xs_lib[i][j][k]["RECIPVEL"] = data['INF_INVV'][index][::2]
+                    self.xs_lib[i][j][k]["CHI"] = data['INF_CHIT'][index][::2]
+                    self.xs_lib[i][j][k]["CHI_D"] = data['INF_CHID'][index][::2]
+                    self.xs_lib[i][j][k]["BETA_EFF"] = data['BETA_EFF'][index][::2]
+                    self.xs_lib[i][j][k]["DECAY_CONSTANT"] = data['LAMBDA'][index][::2]
+                    self.xs_lib[i][j][k]["GTRANSFXS"] = data['INF_SP0'][index][::2]
+
+
+def XS_OUT(fout, xs_dir, mat, temp):
+    for k, v in xs_dir.items():
+        fout.create_dataset(mat+"/"+str(int(temp))+"/"+k, data=v)
+
+
+def TEMP_OUT(fout, mat, temps):
+    fout.create_dataset(mat+"/temp/", data=temps)
+
+
+def READ_INPUT(fin):
+    with io.open(fin) as f:
+        lines = f.readlines()
+    k = 0
+    while k < len(lines):
+        line = lines[k]
+        if '[TITLE]' in line:
+            f = h5py.File(lines[k+1].split()[0], 'w')
+            k += 1
+
+        if '[MAT]' in line:
```

Comment:
Ideally, break this up into separate functions. One for the MAT block, one for the TEMP block, etc.